imageregistry/v1/types.go

```diff
@@ -303,8 +302,32 @@ type ImageRegistryConfigStorageAzure struct {
 	// object.
 	// +optional
 	CloudName string `json:"cloudName,omitempty"`
+	// networkAccess defines the network access level for the storage account.
+	// "Internal" means the storage account will be private, "External" means
+	// the storage account will be publicly accessible.
+	// Defaults to "External".
+	// +kubebuilder:validation:Enum="Internal";"External"
+	// +kubebuilder:default="External"
+	// +optional
+	NetworkAccess AzureNetworkAccessibility `json:"networkAccess,omitempty"`
+	// vnetName is the name of the vnet the registry operates in.
+	// +optional
+	VNetName string `json:"vnetName,omitempty"`
+	// subnetName is the name of the subnet the registry operates in.
+	// +optional
+	SubnetName string `json:"subnetName,omitempty"`
```

Comment:
Is there any specific regex for subnet name? Or, any string is acceptable? Is there a minimal length for the name?

Reply:
Answering here too so it's easier to track: I've set a min/max length, but don't really want to use a regex here. We aren't creating the subnet. Users could give us a valid name with a typo and there would be an error anyway. I don't see the point.
Am I missing something?

Reply:
The more validation the better. Though, if the regex is not known in advance or can change without warning/often, then server-side validation is sufficient. It's just better to fail early if there's a possibility.

Reply:
Turns out the name requirements were nearly identical to the private endpoint name, so I was able to reuse the regex.